### PR TITLE
implementing solarized dark theme

### PR DESCRIPTION
**Pull request by the-gem-dev**
working on the solarized dark theme for gdevelop

solarized2

### CODE REVIEW

**Comment by 4ian** [comment]
Nice :) Could you post more screenshots to see how it looks in the rest of the app? 
Also if you can explain your choices of colours, I guess you've used the one in the solarized dark color swatch? :)

File: newIDE/app/src/UI/Theme/SolarizedDarkTheme/EventsSheet.css [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol' !important;
+  font-size: 14px !important;
+  background-color: #224ea7;
+}
+
+/**
+ * Ensure scroll is only on Y direction, preventing the display
+ * of horizontal scrollbar.
+ */
+.gd-events-sheet-nord-theme .rst__virtualScrollOverride {
```
Comment (4ian) [inline newIDE/app/src/UI/Theme/SolarizedDarkTheme/EventsSheet.css:20]: You'll need to rename everything called xxx-nord-theme to xxx-solarized-dark-theme in all CSS files and update the JavaScript file to set this style name :)

File: newIDE/app/src/UI/Theme/SolarizedDarkTheme/index.js [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    selectedWarningBackgroundColor: '#ffb032',
+  },
+  emptyMessage: {
+    shadowColor: '#000',
+  },
+  logo: {
+    src: 'res/GD-logo-big.png',
+  },
+  mosaicRootClassName: 'mosaic-gd-dark-theme', // See Mosaic.css
+  eventsSheetRootClassName: 'gd-events-sheet-dark-theme', // See EventsSheet.css
```
Comment (4ian) [inline newIDE/app/src/UI/Theme/SolarizedDarkTheme/index.js:94]: Here replace every dark- or nord- by solarized-dark- so that your CSS files rules are used

Comment (the-gem-dev) [reply newIDE/app/src/UI/Theme/SolarizedDarkTheme/index.js:94]: Thank you. Previous I used xxx-solarised-dark-theme but when I did that the css stopped working and I had to spend hours going through the codes in the ui. I never knew it was right in the index folder..

**Comment by the-gem-dev** [comment]
I am not on pc. now but I have a screen shot 
solarized3

**Comment by the-gem-dev** [comment]
Saturation 70% and brightness 90%

**Comment by 4ian** [comment]
Thanks! I meant screenshot of the test of the app, some dialogs, the sprite editor, the events editor, the extension editor, etc... Everything to see how it looks :)

**Comment by the-gem-dev** [comment]
Okay, I am working on the events now I would send screenshot as soon as I am done

File: newIDE/app/src/UI/Theme/SolarizedDarkTheme/Mosaic.css [hunk 3]
```diff
@@ -0,0 +1,46 @@
+/* Mosaic layout */
```
Comment (4ian) [inline newIDE/app/src/UI/Theme/SolarizedDarkTheme/Mosaic.css:1]: Clean this duplicated comment

File: newIDE/app/src/UI/Theme/SolarizedDarkTheme/Table.css [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  border-bottom: 1px solid #3B4252;
+}
+.gd-table-solarized-dark-theme .tableOddRow {
+  background-color: #3B4252;
+}
+.gd-table-solarized-dark-theme .tableEvenRow {
+  background-color: #2E3440;
+}
+.gd-table-solarized-dark-theme .tableSelectedRow {
+  background-color: rgb(6, 172, 218); /*OS X selection*/
```
Comment (4ian) [inline newIDE/app/src/UI/Theme/SolarizedDarkTheme/Table.css:27]: Remove the comment if not true anymore

File: newIDE/app/src/UI/Theme/SolarizedDarkTheme/index.js [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+import './EventsSheet.css';
+import 'react-virtualized/styles.css'; // Styles for react-virtualized Table
+import './Table.css';
+import './Markdown.css';
+
+const almostWhite = '#EEE';
+const lightWhite = '#DDD';
+const notSoWhite = '#CCC';
+const solarizedDark1 = '#047AA6';
+const solarizedDark2 = '#047AA6';
```
Comment (4ian) [inline newIDE/app/src/UI/Theme/SolarizedDarkTheme/index.js:14]: Don't use two varaibles if they are the same

**Comment by 4ian** [comment]
Thanks! :) I added a few comments. Can you list the color from the solarized dark theme that you've used?

File: newIDE/app/src/UI/Theme/SolarizedDarkTheme/Mosaic.css (hunk 3 above)
Comment (the-gem-dev) [reply newIDE/app/src/UI/Theme/SolarizedDarkTheme/Mosaic.css:1]: okay

File: newIDE/app/src/UI/Theme/SolarizedDarkTheme/Table.css (hunk 4 above)
Comment (the-gem-dev) [reply newIDE/app/src/UI/Theme/SolarizedDarkTheme/Table.css:27]: okay

Comment (the-gem-dev) [reply newIDE/app/src/UI/Theme/SolarizedDarkTheme/Table.css:27]: okay

File: newIDE/app/src/UI/Theme/SolarizedDarkTheme/index.js (hunk 5 above)
Comment (the-gem-dev) [reply newIDE/app/src/UI/Theme/SolarizedDarkTheme/index.js:14]: okay

**Comment by the-gem-dev** [comment]
i have a question, where can i find the styles for this.

**Comment by 4ian** [comment]
Use the Chrome color picker to get the Hex number, then search for it in the codebase (in lowercase, then in uppercase, to be sure to find it) ;)

**Comment by the-gem-dev** [comment]
found it, thanks.

**Comment by the-gem-dev** [comment]
i cant seem to find this one
mis

**Comment by the-gem-dev** [comment]
@4ian its been weeks now, you have not said anything about this theme.

**Comment by 4ian** [comment]
Looks mostly good to me :) Let me make a last check and merge this.

**Comment by the-gem-dev** [comment]
Okay I can't wait 😊😊😊😊

**Comment by 4ian** [comment]
Can't spot anything wrong, I'm merging this! Thanks a lot for your work on this :) 👍👍
Going forward it would be interesting to make theme simpler to create and maintain, so I created this: https://github.com/4ian/GDevelop/issues/1918

**Comment by Bouh** [comment]
There is an useless file added in this PR.
`newIDE/Thumbs.db`
I delete it, it's a artefact of thumbnail from Windows.

**Comment by 4ian** [comment]
Thanks @bouh! I forgot about this when merging.

**Pull request merged**